mmv1/third_party/terraform/data_sources/data_source_google_compute_region_instance_template.go.erb

```diff
@@ -0,0 +1,119 @@
+<% autogen_exception -%>
+package google
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+<% if version == "ga" -%>
+	"google.golang.org/api/compute/v1"
+<% else -%>
+	compute "google.golang.org/api/compute/v0.beta"
+<% end -%>
+)
+
+func DataSourceGoogleComputeRegionInstanceTemplate() *schema.Resource {
+	// Generate datasource schema from resource
+	dsSchema := datasourceSchemaFromResourceSchema(ResourceComputeRegionInstanceTemplate().Schema)
+
+	dsSchema["filter"] = &schema.Schema{
+		Type:     schema.TypeString,
+		Optional: true,
+	}
+	dsSchema["most_recent"] = &schema.Schema{
+		Type:     schema.TypeBool,
+		Optional: true,
+	}
```

Comment:
`most_recent` appears to only be settable with `filter` being set. Is this true?

I am thinking we should add a conflicts with between `name `and `filter`. And allow `most_recent` to work if filter is empty. wdyt?